app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CurrencySelectionDialog.kt

```diff
@@ -0,0 +1,169 @@
+                                  it.displayName.equals(searchedCurrency, ignoreCase = true) ||
+                                      it.currencyCode.equals(searchedCurrency, ignoreCase = true)
+                                }
+                            if (newCurrency != null) {
+                              financeViewModel.updateCurrency(newCurrency.currencyCode)
+                              financeViewModel.setShowCurrencyDialogState(false)
+                              isError = false
+                            } else {
+                              isError = true
+                            }
+                          }) {
+                            Icon(
+                                imageVector = Icons.Default.CheckCircle,
+                                contentDescription = Icons.Default.CheckCircle.name,
+                                tint = MaterialTheme.colorScheme.primary)
+                          }
+                    },
+                    maxLines = 1,
+                )
+
+                // Currency list to display, based on user search input
+                LazyColumn(
+                    modifier = Modifier.fillMaxSize(),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.Top) {
+                      // currencies filtered based on text input search
+                      val filteredCurrencies =
+                          currencies
+                              .toList()
+                              .filter {
+                                it.displayName.contains(searchedCurrency, ignoreCase = true) ||
+                                    it.currencyCode.contains(searchedCurrency, ignoreCase = true)
+                              }
+                              .sortedBy { it.displayName }
+                      items(filteredCurrencies) { currency ->
+                        Box(
+                            modifier =
+                                Modifier.fillMaxWidth()
+                                    .padding(horizontal = 20.dp)
+                                    .clip(RoundedCornerShape(5.dp))
+                                    .border(
+                                        1.dp,
+                                        Color.Gray,
+                                        RoundedCornerShape(5.dp),
+                                    )
+                                    .testTag("currencyItem"),
+                            contentAlignment = Alignment.CenterStart) {
+                              Button(
+                                  modifier = Modifier.fillMaxWidth(),
+                                  colors =
+                                      ButtonDefaults.buttonColors(
+                                          containerColor = Color.Transparent),
+                                  onClick = { searchedCurrency = currency.displayName }) {}
```

Comment:
Could remove the {}

Reply:
I have to do it because the button has no content but I need to specify it to make the app compile